Sources/SwiftDiscord/Rest/DiscordRateLimiter.swift

```diff
@@ -152,30 +152,65 @@ public final class DiscordRateLimiter {
 /// Are considered different endpoints
 public struct DiscordRateLimitKey: Hashable {
+	/// URL Parts for the purpose of rate limiting.
+    /// Combine all the parts of the URL into a list of which parts exist
+    /// Ex. /channels/232184444340011009/messages would be represented by [.channels, .channelID, .messages]
+    public struct DiscordRateLimitURLParts: OptionSet {
+        public let rawValue: Int
+
+        static let         guilds = DiscordRateLimitURLParts(rawValue: 1 << 0)
+        static let        guildID = DiscordRateLimitURLParts(rawValue: 1 << 1)
+        static let       channels = DiscordRateLimitURLParts(rawValue: 1 << 2)
+        static let      channelID = DiscordRateLimitURLParts(rawValue: 1 << 3)
+        static let       messages = DiscordRateLimitURLParts(rawValue: 1 << 4)
+        static let messagesDelete = DiscordRateLimitURLParts(rawValue: 1 << 5)
+        static let      messageID = DiscordRateLimitURLParts(rawValue: 1 << 6)
+        static let     bulkDelete = DiscordRateLimitURLParts(rawValue: 1 << 7)
+        static let         typing = DiscordRateLimitURLParts(rawValue: 1 << 8)
+        static let    permissions = DiscordRateLimitURLParts(rawValue: 1 << 9)
+        static let    overwriteID = DiscordRateLimitURLParts(rawValue: 1 << 10)
+        static let        invites = DiscordRateLimitURLParts(rawValue: 1 << 11)
+        static let     inviteCode = DiscordRateLimitURLParts(rawValue: 1 << 12)
+        static let           pins = DiscordRateLimitURLParts(rawValue: 1 << 13)
+        static let       webhooks = DiscordRateLimitURLParts(rawValue: 1 << 14)
+        static let        members = DiscordRateLimitURLParts(rawValue: 1 << 15)
+        static let         userID = DiscordRateLimitURLParts(rawValue: 1 << 16)
+        static let          roles = DiscordRateLimitURLParts(rawValue: 1 << 17)
+        static let         roleID = DiscordRateLimitURLParts(rawValue: 1 << 18)
+        static let           bans = DiscordRateLimitURLParts(rawValue: 1 << 19)
+        static let          users = DiscordRateLimitURLParts(rawValue: 1 << 20)
+        static let      webhookID = DiscordRateLimitURLParts(rawValue: 1 << 21)
+        static let   webhookToken = DiscordRateLimitURLParts(rawValue: 1 << 22)
+        static let          slack = DiscordRateLimitURLParts(rawValue: 1 << 23)
+        static let         github = DiscordRateLimitURLParts(rawValue: 1 << 24)
+
+        public init(rawValue: Int) {
+            self.rawValue = rawValue
+        }
+    }
+
     // MARK: Properties
 
-    /// The guild or channel ID in this endpoint (or "" if neither)
-    public let key: String
+    /// The guild or channel ID in this endpoint (or 0 if neither)
+    public let id: Snowflake
+    public let urlParts: DiscordRateLimitURLParts
 
     /// The hash of the key.
     public var hashValue: Int {
-        return key.hashValue
+        return (urlParts.rawValue &* 33) &+ id.hashValue
```

Comment:
It's from me attempting to look up hash combination functions.  Thinking about it, it's completely unnecessary in this case because its goal is to prevent different combinations of the same numbers from hashing to the same thing in lists of similar data, but there's no way a snowflake and an enum would have similar hashes.